File: examples/v1/rbac/python/__main__.py
import pulumi
import pulumi_pulumiservice.v1 as ps_v1

config = pulumi.Config()
organization_name = config.get("organizationName") or "service-provider-test-org"
name_suffix = config.get("nameSuffix") or "manual"
role_description = config.get("roleDescription") or "Read-only access to stacks, created by the v1 rbac example."

read_only_role = ps_v1.Role(
    "readOnlyRole",
    org_name=organization_name,
    name=f"v1-rbac-read-only-{name_suffix}",
    description=role_description,
    ux_purpose="role",
    details={
        "__type": "PermissionDescriptorAllow",
        "permissions": ["stack:read"],
    },
)

rbac_team = ps_v1.teams.Team(
    "rbacTeam",
    org_name=organization_name,
    name=f"v1-rbac-team-{name_suffix}",
    display_name=f"v1 RBAC Team {name_suffix}",
    description="Team scaffold used by the v1 rbac example.",
)

pulumi.export("roleName", read_only_role.name)
pulumi.export("teamName", rbac_team.name)

File: examples/v1/rbac/go/main.go
package main

import (
	v1 "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v1"
	teams "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v1/teams"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		organizationName := cfg.Get("organizationName")
		if organizationName == "" {
			organizationName = "service-provider-test-org"
		}
		nameSuffix := cfg.Get("nameSuffix")
		if nameSuffix == "" {
			nameSuffix = "manual"
		}
		roleDescription := cfg.Get("roleDescription")
		if roleDescription == "" {
			roleDescription = "Read-only access to stacks, created by the v1 rbac example."
		}

		readOnlyRole, err := v1.NewRole(ctx, "readOnlyRole", &v1.RoleArgs{
			OrgName:     pulumi.String(organizationName),
			Name:        pulumi.String("v1-rbac-read-only-" + nameSuffix),
			Description: pulumi.String(roleDescription),
			UxPurpose:   pulumi.String("role"),
			Details: pulumi.Map{
				"__type":      pulumi.String("PermissionDescriptorAllow"),
				"permissions": pulumi.StringArray{pulumi.String("stack:read")},
			},
		})
		if err != nil {
			return err
		}

		rbacTeam, err := teams.NewTeam(ctx, "rbacTeam", &teams.TeamArgs{
			OrgName:     pulumi.String(organizationName),
			Name:        pulumi.String("v1-rbac-team-" + nameSuffix),
			DisplayName: pulumi.String("v1 RBAC Team " + nameSuffix),
			Description: pulumi.String("Team scaffold used by the v1 rbac example."),
		})
		if err != nil {
			return err
		}

		ctx.Export("roleName", readOnlyRole.Name)
		ctx.Export("teamName", rbacTeam.Name)
		return nil
	})
}
